Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RobotImporter/URDF/VisualsMaker.h"
+#include "RobotImporter/URDF/PrefabMakerUtils.h"
+#include "RobotImporter/URDF/TypeConversions.h"
+
+#include <AtomLyIntegration/CommonFeatures/Material/MaterialComponentBus.h>
+#include <AtomLyIntegration/CommonFeatures/Material/MaterialComponentConstants.h>
+#include <AtomLyIntegration/CommonFeatures/Mesh/MeshComponentBus.h>
+#include <AtomLyIntegration/CommonFeatures/Mesh/MeshComponentConstants.h>
+#include <AzCore/Component/NonUniformScaleBus.h>
+#include <AzCore/Component/TransformBus.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/ToolsComponents/EditorNonUniformScaleComponent.h>
+#include <LmbrCentral/Shape/BoxShapeComponentBus.h>
+#include <LmbrCentral/Shape/CylinderShapeComponentBus.h>
+#include <LmbrCentral/Shape/EditorShapeComponentBus.h>
+#include <LmbrCentral/Shape/SphereShapeComponentBus.h>
+
+namespace ROS2
+{
+    VisualsMaker::VisualsMaker(const AZStd::string& modelPath, const std::map<std::string, urdf::MaterialSharedPtr>& materials)
+        : m_modelPath(modelPath)
+        , m_materials(materials)
+    {
+    }
+
+    void VisualsMaker::AddVisuals(urdf::LinkSharedPtr link, AZ::EntityId entityId)
+    {
+        const AZStd::string typeString = "visual";
+        if (link->visual_array.size() == 0)
+        { // one or zero visuals - element is used
```

Comment:
This comment doesnt match the condition "one or zero visuals". Should the size check for 1 or 0?

Reply:
@adamdbrw please verify if this logic is what you designed.

Reply:
This is URDF parser (urdfdom package) specifics - there is a single element as well as an array for the same thing. Thus, the comment is correct